### PR TITLE
chore: new endtoend test framework

### PR DESCRIPTION
**Pull request by earayu**
## Checklist

-   [x] Tests were added or are not required
-   [x] Documentation was added or is not required

### CODE REVIEW

File: endtoend/framework/clusters/single_node_cluster.go [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+	return s
+}
+
+// SetUpSingleNodeCluster creates a single node cluster with a MySQL and WeScale database.
+// dbName can be an empty string if no database is needed.
+// setupScript can be an empty string if no setup script is needed.
+// cleanupScript can be an empty string if no cleanup script is needed.
+func (s *SingleNodeCluster) SetUp(dbName string, setupScript string, cleanupScript string) error {
+	// Create the database
+	db, err := framework.NewMysqlConnectionPool(s.wescaleHost, s.wescalePort, s.wescaleUser, s.wescalePasswd, "")
```
Comment (newborn22) [inline endtoend/framework/clusters/single_node_cluster.go:80]: need to close `db`

File: endtoend/framework/filter_builder.go [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+	return &FilterBuilder{
+		Name:       filterName,
+		Priority:   "1000",
+		Status:     "ACTIVE",
+		Action:     "wasm_plugin",
+		ActionArgs: fmt.Sprintf("wasm_binary_name=\"%v\"", wasmName),
+	}
+}
+
+// 基本设置方法
```
Comment (newborn22) [inline endtoend/framework/filter_builder.go:49]: Chinese comment

File: endtoend/framework/clusters/single_node_cluster.go (hunk 1 above)
Comment (earayu) [reply endtoend/framework/clusters/single_node_cluster.go:80]: fixed

File: endtoend/framework/filter_builder.go (hunk 2 above)
Comment (earayu) [reply endtoend/framework/filter_builder.go]: fixed

**Review by newborn22 — APPROVED** [review]
LGTM

**Pull request merged**